presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -728,4 +695,87 @@ private PlanNode distinct(PlanNode node)
                 1.0,
                 Optional.empty());
     }
+
+    private class SetOperationPlanner
+    {
+        private final SetOperation node;
+        private ImmutableList.Builder<PlanNode> sources;
+        private ImmutableListMultimap.Builder<Symbol, Symbol> symbolMapping;
+        private Optional<Symbol> outputSampleWeight;
+
+        public SetOperationPlanner(SetOperation node)
+        {
+            this.node = node;
+        }
+
+        public ImmutableList<PlanNode> getSources()
+        {
+            return sources.build();
+        }
+
+        public ImmutableListMultimap<Symbol, Symbol> getSymbolMapping()
+        {
+            return symbolMapping.build();
+        }
+
+        public Optional<Symbol> getOutputSampleWeight()
+        {
+            return outputSampleWeight;
+        }
+
+        public SetOperationPlanner process()
```

Comment:
Instead of introducing a "processor" class for this, move the logic into a separate method in the enclosing class and have it return a "structure" (i.e., only fields, no logic)